PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/ui/SparkSubmissionJobUploadStoragePanel.kt

```diff
@@ -143,6 +152,53 @@ open class SparkSubmissionJobUploadStoragePanel: JPanel(), Disposable {
         var deployStorageTypesModel: ComboBoxModel<SparkSubmitStorageType> by ComboBoxModelDelegated(storageTypeComboBox)
 
         val storageCheckSubject: PublishSubject<StorageCheckEvent> = disposableSubjectOf { PublishSubject.create() }
+
+        fun prepareVFSRoot(uploadRootPath: String?, storageAccount: IHDIStorageAccount?): AzureStorageVirtualFile? {
+            var fileSystem: AzureStorageVirtualFileSystem? = null
+            var account: String? = null
+            var accessKey: String? = null
+            var fsType: VFSSupportStorageType? = null
+            try {
+                when (viewModel.deployStorageTypeSelection) {
+                    SparkSubmitStorageType.DEFAULT_STORAGE_ACCOUNT -> {
+                        when (storageAccount) {
+                            is ADLSGen2StorageAccount -> {
+                                fsType = VFSSupportStorageType.ADLSGen2
+                                account = storageAccount.name
+                                accessKey = storageAccount.primaryKey
+                            }
+                        }
+                    }
+
+                    SparkSubmitStorageType.ADLS_GEN2 -> {
+                        fsType = VFSSupportStorageType.ADLSGen2
+                        val host = URI.create(uploadRootPath).host
+                        val account = host.substring(0, host.indexOf("."))
+                        var accessKey = adlsGen2Card.storageKeyField.text.trim()
+                    }
+
+                    else -> {
+                    }
+                }
+            } catch (ex: IllegalArgumentException) {
```

Comment:
Which codes would throw the `IllegalArgumentException`?

Reply:
URI.create